rotkehlchen_mock/__main__.py

```diff
@@ -0,0 +1,82 @@
+from gevent import monkey
+
+monkey.patch_all()  # isort:skip
+
+import logging
+from http import HTTPStatus
+from typing import Any
+
+import gevent
+from flask import Response
+
+from rotkehlchen.api.rest import RestAPI, api_response
+from rotkehlchen.api.server import APIServer
+from rotkehlchen.args import app_args
+from rotkehlchen.logging import TRACE, RotkehlchenLogsAdapter, add_logging_level, configure_logging
+from rotkehlchen.rotkehlchen import Rotkehlchen
+from rotkehlchen.server import RotkehlchenServer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def get_well_formed_response(*_: Any) -> Response:
+    return api_response(
+        {
+            'result': {},
+        },
+        status_code=HTTPStatus.OK,
+    )
+
+
+def patched_rotkehlchen_server_constructor(self: Any) -> None:
+    # mock entry point inherits all the cmd args of normal entry point
+    arg_parser = app_args(
+        prog='rotkehlchen_mock',
+        description='mocked rotki server',
+    )
+
+    # you can add additional cmd args
+    args = arg_parser.parse_args()
+
+    add_logging_level('TRACE', TRACE)
+    configure_logging(args)
+
+    rotkehlchen = Rotkehlchen(args)
+    rest_api = RestAPI(rotkehlchen)
+    if ',' in args.api_cors:
+        domain_list = [str(domain) for domain in args.api_cors.split(',')]
+    else:
+        domain_list = [str(args.api_cors)]
+
+    self.args = args
+    self.rotkehlchen = rotkehlchen
+    self.stop_event = gevent.event.Event()
+
+    self.api_server = APIServer(
+        rest_api=rest_api,
+        ws_notifier=self.rotkehlchen.rotki_notifier,
+        cors_domain_list=domain_list,
+    )
+
+    # patch all methods except special ones and those managing state of the app
+    functions_to_patch = [
+        member_name
+        for member_name in dir(rest_api)
+        if callable(getattr(rest_api, member_name)) and
+        not member_name.startswith('__') and
+        member_name != 'stop'
+    ]
+
+    for function_name in functions_to_patch:
+        setattr(rest_api, function_name, get_well_formed_response)
```

Comment:
here you are looping twice the methods. Make it a generator instead since you don't need all of them in memory

```suggestion
    functions_to_patch = (
        member_name
        for member_name in dir(rest_api)
        if callable(getattr(rest_api, member_name)) and
        not member_name.startswith('__') and
        member_name != 'stop'
    )

    for function_name in functions_to_patch:
        setattr(rest_api, function_name, get_well_formed_response)
```